lib/public_key/src/pubkey_ocsp.erl

```diff
@@ -0,0 +1,290 @@
+get_public_key_rec(#'OTPCertificate'{tbsCertificate = TbsCert}) ->
+    PKInfo = TbsCert#'OTPTBSCertificate'.subjectPublicKeyInfo,
+    PKInfo#'OTPSubjectPublicKeyInfo'.subjectPublicKey.
+
+%%--------------------------------------------------------------------
+-spec is_responder(tuple(), #'Certificate'{} | #'OTPCertificate'{}) ->
+    boolean().
+%%
+%% Description: Check if is OCSP responder's cert
+%%--------------------------------------------------------------------
+is_responder({byName, Name}, Cert) ->
+    public_key:der_encode('Name', Name) == get_subject_name(Cert);
+is_responder({byKey, Key}, Cert) ->
+    Key == crypto:hash(sha, get_public_key(Cert)).
+
+%%--------------------------------------------------------------------
+-spec otp_cert(#'Certificate'{} | #'OTPCertificate'{} | binary()) ->
+    #'OTPCertificate'{}.
+%%
+%% Description: Convert to #'OTPCertificate'{}
+%%--------------------------------------------------------------------
+otp_cert(#'OTPCertificate'{} = Cert) ->
+    Cert;
+otp_cert(#'Certificate'{} = Cert) ->
+    public_key:pkix_decode_cert(
+        public_key:der_encode('Certificate', Cert), otp);
+otp_cert(CertDer) when is_binary(CertDer) ->
+    public_key:pkix_decode_cert(CertDer, otp).
+
+%%--------------------------------------------------------------------
+-spec get_ocsp_responder_id(#'Certificate'{}) -> binary().
+%%
+%% Description: Get the OCSP responder ID der
+%%--------------------------------------------------------------------
+get_ocsp_responder_id(#'Certificate'{tbsCertificate = TbsCert}) ->
+    public_key:der_encode(
+        'ResponderID', {byName, TbsCert#'TBSCertificate'.subject}).
+
+%%--------------------------------------------------------------------
+-spec get_subject_name(#'Certificate'{} | #'OTPCertificate'{}) -> binary().
+%%
+%% Description: Get the subject der from cert
+%%--------------------------------------------------------------------
+get_subject_name(#'Certificate'{} = Cert) ->
+    get_subject_name(otp_cert(Cert));
+get_subject_name(#'OTPCertificate'{tbsCertificate = TbsCert}) ->
+    public_key:pkix_encode('Name', TbsCert#'OTPTBSCertificate'.subject, otp).
+
+%%--------------------------------------------------------------------
+-spec get_public_key(#'Certificate'{} | #'OTPCertificate'{}) -> binary().
+%%
+%% Description: Get the public key der from cert
+%%--------------------------------------------------------------------
+get_public_key(#'Certificate'{} = Cert) ->
+    get_public_key(otp_cert(Cert));
+get_public_key(#'OTPCertificate'{tbsCertificate = TbsCert}) ->
+    PKInfo = TbsCert#'OTPTBSCertificate'.subjectPublicKeyInfo,
+    PKInfo#'OTPSubjectPublicKeyInfo'.subjectPublicKey.
+
+%%--------------------------------------------------------------------
+-spec get_nonce(boolean()) -> binary() | undefined.
+%%
+%% Description: Get an OCSP nonce
+%%--------------------------------------------------------------------
+get_nonce(false) ->
```

Comment:
This should be `public_ocsp:get_nonce/0`.

Reply:
done